test/gtest/layout_transpose.cpp

```diff
@@ -0,0 +1,654 @@
+/*******************************************************************************
+ *
+ * MIT License
+ *
+ * Copyright (c) 2024 Advanced Micro Devices, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ *******************************************************************************/
+#include <gtest/gtest.h>
+
+#include <cstdlib>
+#include <ctime>
+#include <vector>
+
+#include <boost/optional.hpp>
+#include "../../driver/conv_common.hpp"
+#include <miopen/batched_transpose_sol.hpp>
+#include <miopen/handle.hpp>
+#include <miopen/invoke_params.hpp>
+#include <miopen/invoker.hpp>
+#include <miopen/miopen.h>
+#include <miopen/tensor.hpp>
+#include "../tensor_holder.hpp"
+#include <miopen/tensor_layout.hpp>
+
+#include "driver.hpp"
+#include "random.hpp"
+
+#define ASSERT_HIP_SUCCESS(x) ASSERT_EQ((x), hipSuccess)
+
+namespace batched_transpose {
+
+template <typename T>
+void cpu_ncdhw2ndhwc(T* dst, T* src, uint64_t N, uint64_t C, uint64_t D, uint64_t H, uint64_t W)
+{
+    for(uint64_t i_n = 0; i_n < N; i_n++)
```

Comment:
This file looks very similar to `gpu_nchw_nhwc_transpose.cpp`.
Can we gather identical parts into one header and use it everywhere?

Reply:
gpu_nchw_nhwc_transpose.cpp is just an old copy. I used git rename to change it to layout_transpose, but something went wonky and the old file stuck around temporarily.